Lib/distutils/unixccompiler.py

```diff
@@ -188,7 +188,10 @@ def link(self, target_desc, objects,
                         i = 1
                         while '=' in linker[i]:
                             i += 1
-                    linker[i] = self.compiler_cxx[i]
+                    if linker[i].endswith('ld_so_aix'):
+                        # Linker is ld_so_aix, so compiler is next arg
+                        i += 1
+                    linker = linker[:i] + self.compiler_cxx + linker[i+1:]
```

Comment:
I don't understand this change. Currently, it replaces i-th argument with compiler_cxx[i] (one string), whereas your change inserts the whole compiler_cxx *list* into linker list. Sorry, I don't understand from https://bugs.python.org/issue35198 what is the current command line and what should be the command line with your change. Can you please a short example of command line?

In case of doubt, I would prefer to not change the current code on non-AIX platforms. Something like:

                    if linker[i].endswith('ld_so_aix'):
                        # Linker is ld_so_aix, so compiler is next arg
                        i += 1
                        linker = linker[:i] + self.compiler_cxx + linker[i+1:]
                    else:
                        linker[i] = self.compiler_cxx[i]

Reply:
So my thoughts were that if the C++ compiler includes command line arguments, you'd possibly want those carried to the linker command, but I suppose those should be set by `LDFLAGS` instead.

So maybe two indices instead?
```python
                # rename i to cc_idx
                cc_idx = ld_idx
                if linker[ld_idx].endswith('ld_so_aix'):
                    # AIX platforms uses the ld_so_aix linker script to do linking, whose first argument
                    # is the compiler to use. In that case, we need to replace *that* argument
                    # not the ld_so_aix argument
                    ld_idx += 1
                linker[ld_idx] = self.compiler_cxx[cc_idx]
```

Reply:
I'm thinking also of changing the `endswith` to use `os.path.pasename` to match https://github.com/python/cpython/blob/7f7bc73bd260df2aa4ab82ab561b76a09b7da50c/Lib/distutils/unixccompiler.py#L187